src/optionroms.c

```diff
@@ -326,15 +327,20 @@ init_pcirom(struct pci_device *pci, int isvga, u64 *sources)
              , pci->vendor, pci->device);
     struct romfile_s *file = romfile_find(fname);
     struct rom_header *rom = NULL;
-    if (file)
-        rom = deploy_romfile(file);
-    else if (RunPCIroms > 1 || (RunPCIroms == 1 && isvga))
-        rom = map_pcirom(pci);
-    if (! rom)
-        // No ROM present.
-        return;
-    setRomSource(sources, rom, RS_PCIROM | (u32)pci);
-    init_optionrom(rom, pci->bdf, isvga);
+
+    if (pxen){
+        if((strcmp(file->name, "pci8086,157b.rom") == 0) || (strcmp(file->name, "pci8086,1539.rom") == 0)){
```

Comment:
```
if((strcmp(file->name, "pci8086,157b.rom") == 0) ||
   (strcmp(file->name, "pci8086,1539.rom") == 0) ||
   (strcmp(file->name, "pci10ec,8168.rom") == 0)) {
        if (pxen == 0)
            return;
}
if (file)
...
```
First of all, pci10ec,8168.rom is for apu1.
Secondly, do not break the optionrom execution routine. We should only omit the optionrom execution for given optionrom names in case pxen is 0.